storm-core/src/jvm/org/apache/storm/topology/CheckpointTupleForwarder.java

```diff
@@ -53,6 +53,10 @@
     private long lastTxid = Long.MIN_VALUE;
     private AnchoringOutputCollector collector;
 
+    public CheckpointTupleForwarder() {
```

Comment:
Oh I'm seeing StatefulBoltExecutor extends this class.

IMO StatefulBoltExecutor extending CheckpointTupleForwarder could make us confusing, since StatefulBoltExecutor always call default constructor of CheckpointTupleForwarder, which means that StatefulBoltExecutor should hide all methods which access (directly or indirectly) this.bolt to avoid NPE.

Could we introduce base class extracting common logic from CheckpointTupleForwarder?